packages/flutter/lib/src/rendering/object.dart

```diff
@@ -472,6 +472,11 @@ class PaintingContext {
       childContext._stopRecordingIfNeeded();
     } else {
       if (elevation != 0) {
+        // The drawShadow call doesn't add the region of the shadow to the
+        // picture's bounds, so we draw a hardcoded amount of extra space to
+        // account for the maximum potential area of the shadow.
+        canvas.drawRect(offsetBounds.inflate(20.0),
+                        new Paint()..color=const Color(0));
```

Comment:
At some point Skia should provide a way to calculate the area drawn by the shadow and adjust the picture's size accordingly.  Until that's available, we need to do some other operation that causes the picture to extend its bounds.